src/Mermaid.tsx

```diff
@@ -33,12 +40,40 @@ export type MermaidProps = {
  * @returns The component.
  */
 export const Mermaid = ({ chart, config }: MermaidProps): ReactElement<MermaidProps> => {
-  // init diagram once
+  // Due to Docusaurus not correctly parsing client-side from server-side modules, use the provided workaround
+  // found in the accompanying issue: https://github.com/facebook/docusaurus/issues/4268#issuecomment-783553084
+  if (typeof window === 'undefined') {
+    return <div></div>
+  }
+
+  const html: HTMLHtmlElement = document.querySelector('html')!
+
+  // Watch for changes in theme in the HTML attribute `data-theme`.
+  const [theme, setTheme] = useState<mermaidAPI.Theme>('default')
```

Comment:
The initial theme needs to be based on the data-theme attribute otherwise when the page is loaded in dark theme it will be in the light theme

Reply:
Whoops! Nice catch. I've moved all the theme logic to a new file called `theme.helper.spec.ts`